File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):
    sum = 0
    
    for current_number in range(1, number + 1):
        sum = sum + current_number
    
    return sum ** 2

def sum_of_squares(number):
    squares = 0
    
    for current_number in range(1, number + 1):
        squares = squares + (current_number ** 2)

    return squares

def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"
#include <cmath>

namespace difference_of_squares {
    int square_of_sum(int input) {
        int sum = 0;
        for (int i = 1; i <= input; ++i) {
            sum += i;
        }
        return std::pow(sum, 2);
    }
    int sum_of_squares(int input) {
        int squares = 0;
        for (int i = 1; i <= input; ++i) {
            squares += std::pow(i, 2);
        }
        return squares;
    }
    int difference(int input) {
        return square_of_sum(input) - sum_of_squares(input);
    }
}  // namespace difference_of_squares
